compiler/rustc_hir/src/definitions.rs

```diff
@@ -281,7 +281,7 @@ pub enum DefPathData {
     /// An `impl Trait` type node.
     ImplTrait,
     /// `impl Trait` generated associated type node.
-    ImplTraitAssocTy,
+    ImplTraitAssocTy(Symbol),
```

Comment:
I guess there's no *real* reason why this needs to be a separate def path data. We *could* use `TypeNs` here now, (I guess) as long as we keep the actual name that the RPITIT is lowering to unnameable.